it/it-tests/src/test/java/it/user/RailsExternalAuthenticationTest.java

```diff
@@ -340,6 +345,35 @@ public void update_password_of_technical_user() throws Exception {
     }
   }
 
+  /**
+   * SONAR-7640
+   */
+  @Test
+  public void authentication_with_ws() throws Exception {
+    // Given clean Sonar installation and no users in external system
+    String login = USER_LOGIN;
+    String password = "1234567";
+    Map<String, String> users = Maps.newHashMap();
+
+    // When user created in external system
+    users.put(login + ".password", password);
+    updateUsersInExtAuth(users);
+
+//    assertThat(checkAuthenticationWithWebService(login, password).code()).isEqualTo(200);
+    assertThat(checkAuthenticationWithWebService("wrong", password).code()).isEqualTo(401);
+//    assertThat(checkAuthenticationWithWebService(login, "wrong").code()).isEqualTo(401);
+//    assertThat(checkAuthenticationWithWebService(login, null).code()).isEqualTo(401);
+//    assertThat(checkAuthenticationWithWebService(null, null).code()).isEqualTo(200);
+
+    setServerProperty(orchestrator, "sonar.forceAuthentication", "true");
+
+    assertThat(checkAuthenticationWithWebService(login, password).code()).isEqualTo(200);
+    assertThat(checkAuthenticationWithWebService("wrong", password).code()).isEqualTo(401);
+    assertThat(checkAuthenticationWithWebService(login, "wrong").code()).isEqualTo(401);
+    assertThat(checkAuthenticationWithWebService(login, null).code()).isEqualTo(401);
+    assertThat(checkAuthenticationWithWebService(null, null).code()).isEqualTo(401);
```

Comment:
200 and 401 should be defined in constants, or use `java.net.HttpURLConnection.HTTP_UNAUTHORIZED` and `HTTP_OK`